api/odigos/v1alpha1/destination_types.go

```diff
@@ -33,6 +33,9 @@ type DestinationSpec struct {
 
 // DestinationStatus defines the observed state of Destination
 type DestinationStatus struct {
+	// DestinationError and DestinationStatus
+	DestinationError  string `json:"destinationError,omitempty"`
+	DestinationStatus string `json:"destinationStatus,omitempty"`
```

Comment:
I think the canonical way to report such things in k8s is [via `conditions`](https://maelvls.dev/kubernetes-conditions/).

You can see an example for this [here](https://github.com/keyval-dev/odigos/blob/0cae40107c7468243995abd26583500f28f55905/api/odigos/actions/v1alpha1/deleteattribute_types.go#L42).

Conditions will show up in `kubectl describe` and are idiomatic in k8s, we can store multiple types and pinpoint the error or status to a specific phase and more